src/process.ts

```diff
@@ -5,67 +5,74 @@ import { State, Store } from './Store';
 /**
  * The arguments passed to a `Command`
  */
-export interface CommandRequest<T = any> extends State<T> {
-	payload: any[];
+export interface CommandRequest<T = any, P extends object = object> extends State<T> {
+	payload: P;
 }
 
 /**
  * A command factory interface. Returns the passed command. This provides a way to automatically infer and/or
  * verify the type of multiple commands without explicitly specifying the generic for each command
  */
-export interface CommandFactory<T = any> {
-	(command: Command<T>): Command<T>;
+export interface CommandFactory<T = any, P extends object = object> {
+	<R extends object = P>(command: Command<T, R>): Command<T, R>;
```

Comment:
Looking through the docs, my first thought was that it would make sense to let the generic type for the payload be specific on the call to create the command as well as on the command factory itself. It appears you can, but the docs seem to show multiple factories being created instead of using this pattern. Is there any reason for that?

Reply:
Sure, I can add example for that.